blinka_displayio_pygamedisplay.py

```diff
@@ -247,67 +219,73 @@ def rgb_to_surface(buff, size):
                 bytes(buffer), (subrectangle.width(), subrectangle.height())
             )
             self._pygame_screen.blit(image_surface, (subrectangle.x1, subrectangle.y1))
-            # pygame.display.flip()
-            # time.sleep(0.1)
-
         return True
 
+    def _refresh_display(self):
+        """override base-class"""
+        super()._refresh_display()
+        pygame.display.flip()
+        self._last_refresh = time.monotonic()
+        self._refresh_pending = False
+
     def _write(self, command, data):
         pass
         # don't need to write to anything
 
     def _release(self):
-        self._pygame_display_tevent.set()
-        self._pygame_display_thread.join()
         pygame.quit()
 
-    def refresh(self, *, target_frames_per_second=60, minimum_frames_per_second=1):
-        """
-        While normal display-objects call this method also within a refresh
-        loop, this implementation uses this method only for explicit updates.
-        Note that we cannot just call the update-logic directly, since
-        the pygame-display was created on another thread.
-        """
-        # pylint: disable=no-member, unused-argument, protected-access
-        # print("inside refresh()")
-        # if not self._auto_refresh:
-        #     self._pygame_display_force_update = True
-
-        self._refresh_display()
-        self._pygame_display_force_update = False
-        pygame.display.flip()
-
     def _get_refresh_areas(self) -> list[Area]:
         areas = []
         areas.append(self._core.area)
         return areas
 
-    def check_quit(self):
+    def check_quit(self, delay=0.05):
         """
         Check if the quit button on the window is being pressed.
+
+        delay - add a delay to reduce CPU load
         """
         try:
             for event in pygame.event.get():
-                if event.type == pygame.QUIT:
+                if (
+                    event.type == _DISPLAYIO_EVENT
+                    and event.code == _DISPLAYIO_EVENT_CODE_REFRESH
+                ):
+                    self._refresh_display()
+                elif event.type in [pygame.QUIT, pygame.WINDOWCLOSE]:
                     # stop and leave method
-                    # self._pygame_display_tevent.set()
-                    # self._pygame_display_thread.join()
                     pygame.quit()
+                    self._pygame_screen = None
                     return True
+                elif event.type in _PYGAME_REDRAW_EVENTS:
+                    # force refresh even if auto_refresh == False
+                    self._refresh_display()
         except pygame.error:
             print("pygame error during check_quit()")
             print(traceback.format_exc())
             return True
+        time.sleep(delay)
         return False
 
-    def event_loop(self, interval=None, on_time=None, on_event=None, events=None):
+    def event_loop(
```

Comment:
Does this function get used somehow? I tried adding print statements to this and they never get printed out. I also tried commenting this entire `event_loop()` function out and running one of the examples and it still seems to work the same.